test/integration/edition_edit_test.rb

```diff
@@ -35,4 +31,59 @@ class EditionEditTest < IntegrationTest
     assert page.has_text?("Related external links")
     assert page.has_text?("Unpublish")
   end
+
+  context "#metadata" do
+    context "#draft" do
+      setup do
+        click_link("Metadata")
+      end
+
+      should "'Metadata' header and an update button" do
+        within :css, ".gem-c-heading" do
+          assert page.has_text?("Metadata")
+        end
+        assert page.has_button?("Update")
+      end
+
+      should "show slug input box prefilled" do
+        assert page.has_text?("Slug")
+        assert page.has_text?("If you change the slug of a published page, the old slug will automatically redirect to the new one.")
+        assert page.has_field?("artefact[slug]", with: /slug/)
+      end
+
+      should "update and show success message" do
+        fill_in "artefact[slug]", with: "changed-slug"
+        choose("Welsh")
+        click_button("Update")
+
+        assert find(".gem-c-radio input[value='cy']").checked?
+        assert page.has_text?("Metadata has successfully updated")
+        assert page.has_field?("artefact[slug]", with: "changed-slug")
+      end
+    end
+
+    context "#published" do
```

Comment:
Similar to `draft`, `published` is not the name of a method being tested, so we shouldn't have the # prefix in the context name. Might be worth elaborating a bit in the context name as well, for example, "viewing a published edition".